README.md

```diff
@@ -1,39 +1,362 @@
+```dart
+  var logger = Logger();
+  var reviewConditionsBuilder =
+      ReviewConditionsBuilderImplementation<CustomReviewSettings>()
+          .minimumJokesFavorited(3);
+
+  var reviewSettingsSource =
+    GetIt.I.registerSingleton<ReviewSettingsSource<CustomReviewSettings>>(
+      MemoryReviewSettingsSource<CustomReviewSettings>(
+        () => const CustomReviewSettings(),
+      ),
+    );
+
+  GetIt.I.registerSingleton(
+    ReviewService<CustomReviewSettings>(
+      logger: logger,
+      reviewPrompter: ReviewPrompter(logger: logger),
+      reviewSettingsSource: reviewSettingsSource,
+      reviewConditionsBuilder: reviewConditionsBuilder,
+    ),
+  );
+```
+
+Notes:
+  1. You need to specify again that you want the conditions builder the settings source and the service to use your CustomReviewSettings and not the generic.  
+  2. We recommend that you define your own interface that wraps `ReviewService<CustomReviewSettings>` to make the usage code leaner and ease any potential refactorings.
+  ```dart
+    /// This interface wraps ReviewService<CustomReviewSettings> so that you don't have to repeat the generic parameter everywhere that you would use the review service.
+    /// In other words, you should use this interface in the app instead of ReviewService<CustomReviewSettings> because it's leaner.
+    class CustomReviewService implements ReviewService<CustomReviewSettings> {
+      late ReviewService<CustomReviewSettings> _reviewService;
+
+      CustomReviewService(ReviewService<CustomReviewSettings> reviewService) {
+        _reviewService = reviewService;
+      }
 
-{Instructions to quickly get started using the project: pre-requisites, packages
-to install, sample code, etc.}
+      @override
+      Future<bool> getAreConditionsSatisfied() async {
+        return await _reviewService.getAreConditionsSatisfied();
+      }
+
+      @override
+      Future<void> tryRequestReview() async {
+        await _reviewService.tryRequestReview();
+      }
+
+      @override
+      Future<void> updateReviewSettings(
+        CustomReviewSettings Function(CustomReviewSettings p1) updateFunction,
+      ) async {
+        await _reviewService.updateReviewSettings(updateFunction);
+      }
+    }
+  ```
+
+After creating your own implementation of the ReviewService you just need to register it like this :
+```dart
+  GetIt.I.registerSingleton(
+    CustomReviewService(
+      ReviewService<CustomReviewSettings>(
+        logger: logger,
+        reviewPrompter: ReviewPrompter(logger: logger),
+        reviewSettingsSource: reviewSettingsSource,
+        reviewConditionsBuilder: reviewConditionsBuilder,
+      ),
+    ),
+  );
+```
+
+And use it like that :
+```dart
+  await GetIt.I.get<CustomReviewService>().trackFavoriteJokesCount();
+```
+
+now you're all set to track how many jokes are favorited in your app and prompt for a review once your review conditions are meant (3 favorite jokes in this case).
```

Comment:
Capital letter at the start of sentences.

Reply:
Updated.